web/controllers/game_controller.ex

```diff
@@ -0,0 +1,11 @@
+defmodule Yggdrasil.GameController do
+  use Yggdrasil.Web, :controller
+
+  alias Yggdrasil.Game
+
+  def index(conn, _params) do
+    games = Repo.all Game
```

Comment:
OK, I think I'm in pretty much complete agreement with your examples. Exactly, you can't push too much behind the API wall or you'll end up with overly specific functions. So the question comes down to direct repo access. It offers any other sub-systems complete flexibility in how to retrieve data, but to your point we could end up with messy code.

Perhaps as convention we allow queries to the repo only at first? Other APIs are driven off of the struct returned and our module APIs. It could also be convention that you can use the repo directly until it's not just straight access, then logic needs to be pushed down. So if all you really needed to do was straight CRUD then fine, but after that you gotta push the logic down into the modules. Hell the repo is public anyway and usage is only effecting us.